test/snapshot/sampleProjects/destructiveCompSets/snapshots.test.ts

```diff
@@ -0,0 +1,159 @@
+
+  // The directory where metadata is converted as part of testing
+  const testOutput = path.join(testDir, 'testOutput');
+
+  /** Return only the files involved in the conversion */
+  const getConvertedFilePaths = async (outputDir: string): Promise<string[]> =>
+    dirEntsToPaths(
+      await fs.promises.readdir(outputDir, {
+        recursive: true,
+        withFileTypes: true,
+      })
+    );
+
+  it('1 pre-destructive and 1 deployment', async () => {
+    const cs = await ComponentSetBuilder.build({
+      metadata: {
+        metadataEntries: ['ApexClass:OneClass'],
+        destructiveEntriesPre: ['ApexClass:RedClass'],
+        directoryPaths: [sourceDir],
+      },
+      projectDir: testDir,
+    });
+
+    const pre1TestOutputDir = path.join(testOutput, 'pre1');
+
+    await new MetadataConverter().convert(cs, 'metadata', {
+      type: 'directory',
+      outputDirectory: pre1TestOutputDir,
+      genUniqueDir: false,
+    });
+
+    const convertedFiles = await getConvertedFilePaths(pre1TestOutputDir);
+    for (const file of convertedFiles) {
+      await fileSnap(file, testDir);
+    }
+    dirsAreIdentical(path.join(snapshotsDir, 'testOutput', 'pre1'), pre1TestOutputDir);
+  });
+
+  it('2 pre-destructive and 2 deployments', async () => {
+    const cs = await ComponentSetBuilder.build({
+      metadata: {
+        metadataEntries: ['ApexClass:OneClass', 'ApexClass:TwoClass'],
+        destructiveEntriesPre: ['ApexClass:RedClass', 'ApexClass:BlueClass'],
+        directoryPaths: [sourceDir],
+      },
+      projectDir: testDir,
+    });
+
+    const pre2TestOutputDir = path.join(testOutput, 'pre2');
+
+    await new MetadataConverter().convert(cs, 'metadata', {
+      type: 'directory',
+      outputDirectory: pre2TestOutputDir,
+      genUniqueDir: false,
+    });
+
+    const convertedFiles = await getConvertedFilePaths(pre2TestOutputDir);
+    for (const file of convertedFiles) {
+      await fileSnap(file, testDir);
+    }
+    dirsAreIdentical(path.join(snapshotsDir, 'testOutput', 'pre2'), pre2TestOutputDir);
+  });
+
+  it('1 post-destructive and 1 deployment', async () => {
+    const cs = await ComponentSetBuilder.build({
+      metadata: {
+        metadataEntries: ['ApexClass:OneClass'],
+        destructiveEntriesPost: ['ApexClass:RedClass'],
+        directoryPaths: [sourceDir],
+      },
+      projectDir: testDir,
+    });
+
+    const postTestOutputDir = path.join(testOutput, 'post1');
+
+    await new MetadataConverter().convert(cs, 'metadata', {
+      type: 'directory',
+      outputDirectory: postTestOutputDir,
+      genUniqueDir: false,
+    });
+
+    const convertedFiles = await getConvertedFilePaths(postTestOutputDir);
+    for (const file of convertedFiles) {
+      await fileSnap(file, testDir);
+    }
+    dirsAreIdentical(path.join(snapshotsDir, 'testOutput', 'post1'), postTestOutputDir);
+  });
+
+  it('2 post-destructive and 2 deployments', async () => {
```

Comment:
should/could we add a test covering 1+ pre or post destructive changes, without any constructive? Seems like it'll be working - but would be good to have for future changes

Reply:
or maybe we don't want to support that?  if the CLIs not adding new flags to use this, I'm not sure if other SDR consumers would like that ability

Reply:
I can add a test for those cases.  👍